File: src/LifeQuest.Web/src/app/core/labels/labels.ts
import {
  CostBand,
  Difficulty,
  DiscoveryRadius,
  NotificationPreference,
  PhysicalEffort,
  LifeCategory,
  QuestStatus,
  QuestType,
  ScoreBreakdown,
  SkipReason,
} from '../api/models';
import { IconName } from '../../ui/icon';
import { t } from '../i18n/i18n';

/*
 * Etiketler o anki dilin sözlüğünden getter'larla okunur: şablonda ya da computed içinde okunduklarında dil
 * değişince kendiliğinden yenilenirler. Yapı (ikon, sıra, renk) burada, metin core/i18n/sections/labels.ts'te.
 */

export interface CategoryMeta {
  readonly label: string;
  icon: IconName;
  /** CSS değişken adı öneki: var(--cat-explorer), var(--cat-explorer-ink) */
  token: string;
}

const CATEGORY_ICONS: Record<LifeCategory, { icon: IconName; token: string }> = {
  Explorer: { icon: 'compass', token: 'explorer' },
  Culture: { icon: 'landmark', token: 'culture' },
  Learning: { icon: 'book', token: 'learning' },
  Social: { icon: 'users', token: 'social' },
  Fitness: { icon: 'activity', token: 'fitness' },
  Creativity: { icon: 'palette', token: 'creativity' },
};

export const CATEGORY_ORDER: LifeCategory[] = ['Explorer', 'Culture', 'Learning', 'Social', 'Fitness', 'Creativity'];

export const CATEGORIES: Record<LifeCategory, CategoryMeta> = Object.fromEntries(
  CATEGORY_ORDER.map((c) => [c, { ...CATEGORY_ICONS[c], get label() { return t().labels.categories[c]; } }]),
) as Record<LifeCategory, CategoryMeta>;

export const CATEGORY_DESCRIPTIONS: Record<LifeCategory, string> = lookup(CATEGORY_ORDER, (c) => t().labels.categoryDescriptions[c]);

export const COST_ORDER: CostBand[] = ['Free', 'Low', 'Medium', 'High'];

export const COST_LABELS: Record<CostBand, { readonly short: string; readonly label: string; readonly hint: string }> =
  Object.fromEntries(COST_ORDER.map((c) => [c, {
    get short() { return t().labels.cost[c].short; },
    get label() { return t().labels.cost[c].label; },
    get hint() { return t().labels.cost[c].hint; },
  }])) as Record<CostBand, { short: string; label: string; hint: string }>;

const RADIUS_ICONS: Record<DiscoveryRadius, IconName> = { Chill: 'leaf', Explore: 'compass', SurpriseMe: 'sparkles' };

export const RADIUS_LABELS: Record<DiscoveryRadius, { readonly label: string; readonly description: string; icon: IconName }> =
  Object.fromEntries((['Chill', 'Explore', 'SurpriseMe'] as const).map((r) => [r, {
    icon: RADIUS_ICONS[r],
    get label() { return t().labels.radius[r].label; },
    get description() { return t().labels.radius[r].description; },
  }])) as Record<DiscoveryRadius, { label: string; description: string; icon: IconName }>;

export const QUEST_TYPE_LABELS: Record<QuestType, string> =
  lookup<QuestType>(['Daily', 'Weekly', 'Adventure', 'Epic'], (q) => t().labels.questTypes[q]);

export const DIFFICULTY_LABELS: Record<Difficulty, string> =
  lookup<Difficulty>(['Easy', 'Medium', 'Hard', 'Heroic'], (d) => t().labels.difficulties[d]);

export const STATUS_LABELS: Record<QuestStatus, string> =
  lookup<QuestStatus>(['Offered', 'Accepted', 'Completed', 'Skipped', 'Expired'], (s) => t().labels.statuses[s]);

export const SKIP_REASONS: { readonly value: SkipReason; readonly label: string; readonly hint: string }[] =
  (['NotInterested', 'TooExpensive', 'NoTime', 'TooFar', 'NotToday', 'Other'] as const).map((value) => ({
    value,
    get label() { return t().labels.skipReasons[value].label; },
    get hint() { return t().labels.skipReasons[value].hint; },
  }));

export interface ScoreComponentMeta {
  key: keyof Omit<ScoreBreakdown, 'total'>;
  readonly label: string;
  penalty: boolean;
}

export const SCORE_COMPONENTS: ScoreComponentMeta[] = (
  [
    ['interest', false], ['novelty', false], ['context', false], ['goalFit', false], ['diversity', false],
    ['feedbackFit', false], ['repetition', true], ['friction', true], ['risk', true],
  ] as const
).map(([key, penalty]) => ({ key, penalty, get label() { return t().labels.scoreComponents[key]; } }));

export const WEEKLY_TIME_OPTIONS: { minutes: number; readonly label: string; readonly short: string; readonly hint: string }[] =
  ([120, 300, 600, 900] as const).map((minutes) => ({
    minutes,
    get label() { return t().labels.weeklyTime[minutes].label; },
    get short() { return t().labels.weeklyTime[minutes].short; },
    get hint() { return t().labels.weeklyTime[minutes].hint; },
  }));

export const EFFORT_LABELS: Record<PhysicalEffort, { readonly label: string; readonly hint: string }> =
  Object.fromEntries((['None', 'Light', 'Moderate', 'Vigorous'] as const).map((e) => [e, {
    get label() { return t().labels.effort[e].label; },
    get hint() { return t().labels.effort[e].hint; },
  }])) as Record<PhysicalEffort, { label: string; hint: string }>;

/** Profilde seçilebilen efor üst sınırı (erişilebilirlik / hareket kısıtı). */
export const EFFORT_LIMIT_OPTIONS: { value: PhysicalEffort; readonly label: string; readonly hint: string }[] =
  (['Light', 'Moderate', 'Vigorous'] as const).map((value) => ({
    value,
    get label() { return t().labels.effortLimit[value].label; },
    get hint() { return t().labels.effortLimit[value].hint; },
  }));

export const NOTIFICATION_LABELS: Record<NotificationPreference, string> =
  lookup<NotificationPreference>(['Off', 'WeeklySummary'], (n) => t().labels.notifications[n]);

/** Anahtarları sabit, değerleri her okumada sözlükten gelen salt okunur harita. */
function lookup<K extends string>(keys: readonly K[], read: (key: K) => string): Record<K, string> {
  const target = {} as Record<K, string>;
  for (const key of keys) Object.defineProperty(target, key, { get: () => read(key), enumerable: true });
  return target;
}

File: src/LifeQuest.Mobile.Core/Formatting/Labels.cs
using LifeQuest.Mobile.Core.Api;
using LifeQuest.Mobile.Core.Localization;

namespace LifeQuest.Mobile.Core.Formatting;

/// <summary>Kategori görsel kimliği: ikon adı ve renk anahtarı (Resources/Styles'taki Cat{Key} renkleri).</summary>
public sealed record CategoryMeta(LifeCategory Category, string Icon, string ColorKey)
{
    public string Label => Localizer.Instance.S.Labels.Categories[Category.ToString()];
    public string Description => Localizer.Instance.S.Labels.CategoryDescriptions[Category.ToString()];
}

public sealed record Option<T>(T Value, string Label, string? Hint = null, string? Icon = null);

/// <summary>Enum → etiket köprüsü (web: core/labels/labels.ts). Metinler sözlükten, yapı (ikon, sıra) buradan gelir.</summary>
public static class Labels
{
    private static LabelsStrings L => Localizer.Instance.S.Labels;

    public static readonly LifeCategory[] CategoryOrder =
        [LifeCategory.Explorer, LifeCategory.Culture, LifeCategory.Learning, LifeCategory.Social, LifeCategory.Fitness, LifeCategory.Creativity];

    public static CategoryMeta Category(LifeCategory category) => category switch
    {
        LifeCategory.Explorer => new(category, "compass", "Explorer"),
        LifeCategory.Culture => new(category, "landmark", "Culture"),
        LifeCategory.Learning => new(category, "book", "Learning"),
        LifeCategory.Social => new(category, "users", "Social"),
        LifeCategory.Fitness => new(category, "activity", "Fitness"),
        _ => new(category, "palette", "Creativity")
    };

    public static ILabelsCostItem Cost(CostBand cost) => L.Cost[cost.ToString()];

    public static ILabelsRadiusItem Radius(DiscoveryRadius radius) => L.Radius[radius.ToString()];

    public static string RadiusIcon(DiscoveryRadius radius) => radius switch
    {
        DiscoveryRadius.Chill => "leaf",
        DiscoveryRadius.Explore => "compass",
        _ => "sparkles"
    };

    public static string QuestType(QuestType type) => L.QuestTypes[type.ToString()];
    public static string Difficulty(Difficulty difficulty) => L.Difficulties[difficulty.ToString()];
    public static string Status(QuestStatus status) => L.Statuses[status.ToString()];
    public static ILabelsSkipReasonsItem SkipReason(SkipReason reason) => L.SkipReasons[reason.ToString()];
    public static ILabelsEffortItem Effort(PhysicalEffort effort) => L.Effort[effort.ToString()];
    public static string Notification(NotificationPreference preference) => L.Notifications[preference.ToString()];

    public static readonly CostBand[] CostOrder = [CostBand.Free, CostBand.Low, CostBand.Medium, CostBand.High];
    public static readonly DiscoveryRadius[] RadiusOrder = [DiscoveryRadius.Chill, DiscoveryRadius.Explore, DiscoveryRadius.SurpriseMe];
    public static readonly SkipReason[] SkipReasons =
        [Api.SkipReason.NotInterested, Api.SkipReason.TooExpensive, Api.SkipReason.NoTime, Api.SkipReason.TooFar, Api.SkipReason.NotToday, Api.SkipReason.Other];

    /// <summary>Haftalık zaman seçenekleri (dakika).</summary>
    public static readonly int[] WeeklyTimeOptions = [120, 300, 600, 900];

    public static ILabelsWeeklyTimeItem WeeklyTime(int minutes) => L.WeeklyTime[minutes.ToString(System.Globalization.CultureInfo.InvariantCulture)];

    /// <summary>Profilde seçilebilen efor üst sınırı.</summary>
    public static readonly PhysicalEffort[] EffortLimits = [PhysicalEffort.Light, PhysicalEffort.Moderate, PhysicalEffort.Vigorous];

    public static ILabelsEffortLimitItem EffortLimit(PhysicalEffort effort) => L.EffortLimit[effort.ToString()];

    /// <summary>Skor bileşenleri: anahtar, etiket, ceza mı.</summary>
    public static IReadOnlyList<(string Key, string Label, bool Penalty, Func<ScoreBreakdown, double> Value)> ScoreComponents =>
    [
        ("interest", L.ScoreComponents.Interest, false, s => s.Interest),
        ("novelty", L.ScoreComponents.Novelty, false, s => s.Novelty),
        ("context", L.ScoreComponents.Context, false, s => s.Context),
        ("goalFit", L.ScoreComponents.GoalFit, false, s => s.GoalFit),
        ("diversity", L.ScoreComponents.Diversity, false, s => s.Diversity),
        ("feedbackFit", L.ScoreComponents.FeedbackFit, false, s => s.FeedbackFit),
        ("repetition", L.ScoreComponents.Repetition, true, s => s.Repetition),
        ("friction", L.ScoreComponents.Friction, true, s => s.Friction),
        ("risk", L.ScoreComponents.Risk, true, s => s.Risk)
    ];

    /// <summary>Hava ikonu WMO koduna göre (web weather-chip ile aynı).</summary>
    public static string WeatherIcon(int code) => code >= 51 ? "cloud-rain" : code == 0 ? "sun" : "cloud";
}
